spec/services/ab_test_bucket_spec.rb

```diff
@@ -0,0 +1,123 @@
+require 'rails_helper'
+
+describe AbTestBucket do
+  let(:foo_percent) { 30 }
+  let(:bar_percent) { 20 }
+  let(:baz_percent) { 40 }
+  let(:default_percent) { 10 }
+  let(:acceptable_delta) { 5 }
+  let(:subject) do
+    AbTestBucket.new(buckets: { foo: foo_percent, bar: bar_percent, baz: baz_percent })
+  end
+
+  context 'configured with buckets adding up to less than 100 percent' do
+    it 'splits random uuids into the buckets to within an acceptable delta percent' do
```

Comment:
This spec is not guaranteed to pass, but running it 10000 times, it only failed 22 times (see the following, skipped, spec). I do think it is important to see that it comes close to the percentages desired so I'm leaving it in

Reply:
In general I'm not a fan of testing randomness like that... I'd stub `Digest::SHA256.hexdigest(x)` and hardcode few user IDs to return fixed values like `000000`, `fffffff`, etc

Reply:
love it, thanks!